soc/arm/nxp_imx/rt/power_rt11xx.c

```diff
@@ -0,0 +1,337 @@
+	GPC_STBY_ConfigStandbyTransitionStep(GPC_STBY_CTRL,
+		kGPC_STBY_PllOut, &step_cfg);
+	GPC_STBY_ConfigStandbyTransitionStep(GPC_STBY_CTRL,
+		kGPC_STBY_LpcgOut, &step_cfg);
+}
+
+static void gpc_configure_interrupts(void)
+{
+	uint8_t i;
+	uint32_t irq = DT_IRQN(DT_INST(0, nxp_gpt_hw_timer));
+
+	/* Disable all GPC interrupt sources */
+	for (i = 0; i < GPC_CPU_MODE_CTRL_CM_IRQ_WAKEUP_MASK_COUNT; i++) {
+		GPC_CPU_MODE_CTRL->CM_IRQ_WAKEUP_MASK[i] |= 0xFFFFFFFF;
+	}
+
+	/* Enable GPT interrupt source for GPC- this is system timer */
+	GPC_CM_EnableIrqWakeup(GPC_CPU_MODE_CTRL, irq, true);
+}
+
+/* Initializes configuration for the GPC */
+static void gpc_init(void)
+{
+	/* Setup GPC set point mappings */
+	gpc_set_core_mappings();
+	/* Setup GPC set point transition flow */
+	gpc_set_transition_flow();
+	/* Allow GPC to disable ROSC */
+	GPC_SET_POINT_CTRL->SP_ROSC_CTRL = ~OSC_RC_16M_STBY_VAL;
+	/* Setup GPC interrupts */
+	gpc_configure_interrupts();
+}
+
+/* Initializes DCDC converter with power saving settings */
+static void dcdc_init(void)
+{
+	dcdc_config_t dcdc_config;
+	dcdc_setpoint_config_t dcdc_setpoint_config;
+
+	dcdc_buck_mode_1P8_target_vol_t buck1_8_voltage[16] =
+		DCDC_1P8_BUCK_MODE_CONFIGURATION_TABLE;
+	dcdc_buck_mode_1P0_target_vol_t buck1_0_voltage[16] =
+		DCDC_1P0_BUCK_MODE_CONFIGURATION_TABLE;
+	dcdc_standby_mode_1P8_target_vol_t standby1_8_voltage[16] =
+		DCDC_1P8_STANDBY_MODE_CONFIGURATION_TABLE;
+	dcdc_standby_mode_1P0_target_vol_t standby1_0_voltage[16] =
+		DCDC_1P0_STANDBY_MODE_CONFIGURATION_TABLE;
+
+
+	DCDC_BootIntoDCM(DCDC);
+
+	dcdc_setpoint_config.enableDCDCMap = DCDC_ONOFF_SP_VAL;
+	dcdc_setpoint_config.enableDigLogicMap = DCDC_DIG_ONOFF_SP_VAL;
+	dcdc_setpoint_config.lowpowerMap = DCDC_LP_MODE_SP_VAL;
+	dcdc_setpoint_config.standbyMap = DCDC_ONOFF_STBY_VAL;
+	dcdc_setpoint_config.standbyLowpowerMap = DCDC_LP_MODE_STBY_VAL;
+	dcdc_setpoint_config.buckVDD1P8TargetVoltage = buck1_8_voltage;
+	dcdc_setpoint_config.buckVDD1P0TargetVoltage = buck1_0_voltage;
+	dcdc_setpoint_config.standbyVDD1P8TargetVoltage = standby1_8_voltage;
+	dcdc_setpoint_config.standbyVDD1P0TargetVoltage = standby1_0_voltage;
+	DCDC_SetPointInit(DCDC, &dcdc_setpoint_config);
+
+	DCDC_GetDefaultConfig(&dcdc_config);
+	dcdc_config.controlMode = kDCDC_SetPointControl;
+	DCDC_Init(DCDC, &dcdc_config);
+}
+
+static void system_enter_sleep(gpc_cpu_mode_t gpc_mode)
+{
+	__ASSERT_NO_MSG(gpc_mode != kGPC_RunMode);
+
+	if (gpc_mode == kGPC_WaitMode) {
+		/* Clear SLEEPDEEP bit to enter WAIT mode*/
+		SCB->SCR &= ~SCB_SCR_SLEEPDEEP_Msk;
+	} else {
+		/* Set SLEEPDEEP bit to enter STOP mode */
+		SCB->SCR |= SCB_SCR_SLEEPDEEP_Msk;
+	}
+	/* FIXME: When this function is entered the Kernel has disabled
+	 * interrupts using BASEPRI register. This is incorrect as it prevents
+	 * waking up from any interrupt which priority is not 0. Work around the
+	 * issue and disable interrupts using PRIMASK register as recommended
+	 * by ARM.
+	 */
```

Comment:
The thing is that we can't be interrupted during the sleep process in the idle
thread until we call arch specific code. Then its is up to the target set up
which interruption should wake up the system.

Reply:
So to clarify, the kernel is disabling interrupts to avoid the sleep process being interrupted, but once SOC specific power management code is called the SOC is free to enable interrupts again? If so, I'm assuming that I should just change the comment here to reflect the correct reason that the BASEPRI register needs to be set.